.changeset/itchy-items-hang.md

```diff
@@ -0,0 +1,5 @@
+---
+"react-router-dom": patch
+---
+
+Fix `NavLink` `active` logic when `to` location has a trailing slash
```

Comment:
@bilalk711 This seems not what you described in the ticket?
Can you please remove this?

If this is meant to be what's solved, can you please elaborate on why changing `substr` to `substring` fixes it?